tests/tests.py

```diff
@@ -73,13 +76,17 @@ def test_login_returns_serialized_token(self):
         self.assertNotIn(username_field, response.data)
 
     def test_login_returns_serialized_token_and_username_field(self):
-        self.assertEqual(AuthToken.objects.count(), 0)
-        url = reverse('knox_login')
-        self.client.credentials(
-            HTTP_AUTHORIZATION=get_basic_auth_header(self.username, self.password)
-        )
-        knox_settings.USER_SERIALIZER = UserSerializer
-        response = self.client.post(url, {}, format='json')
+
+        with override_settings(REST_KNOX=token_user_limit_knox):
+            reload_module(views)
+            self.assertEqual(AuthToken.objects.count(), 0)
+            url = reverse('knox_login')
+            self.client.credentials(
+                HTTP_AUTHORIZATION=get_basic_auth_header(self.username, self.password)
+            )
+            knox_settings.USER_SERIALIZER = UserSerializer
```

Comment:
This line is not necesarry since you already overwrote the setting

Reply:
Thats where you're mistaken my friend. `knox_settings.USER_SERIALIZER = UserSerializer` is indeed still needed otherwise `USER_SERIALIZER` is `None` or I need to reload some other module perhaps